gensim/scripts/segment_wiki.py

```diff
@@ -122,9 +120,13 @@ def segment_and_write_all_articles(file_path, output_file, min_article_character
                 output_data["section_texts"].append(section_content)
             if (idx + 1) % 100000 == 0:
                 logger.info("processed #%d articles (at %r now)", idx + 1, article_title)
-            outfile.write(json.dumps(output_data) + "\n")
+            if output_file is None:
+                sys.stdout.write(json.dumps(output_data) + "\n")
+            else:
+                outfile.write((json.dumps(output_data) + "\n").encode())
```

Comment:
Definitely not: always use explicit encoding!

In this case, the output must be utf8.

Reply:
I'd prefer to write utf8 even to stdout (`sys.stdout.buffer`), because that's the script contract -- that's what we tell users we output. It's not a special case.

Reply:
```
>>> help(str.encode)
Help on method_descriptor:

encode(...)
    S.encode(encoding='utf-8', errors='strict') -> bytes
```


On the Python2 too.

Reply:
No, it's not:

```python
u"ř".encode()

UnicodeEncodeError: 'ascii' codec can't encode character u'\u0159' in position 0: ordinal not in range(128)
```